src/main/java/com/example/hbs/service/BookingService.java

```diff
@@ -1,27 +1,55 @@
 package com.example.hbs.service;
 
+import com.example.hbs.dto.BookingRequestDto;
+import com.example.hbs.dto.BookingResponseDto;
 import com.example.hbs.model.Booking;
-import com.example.hbs.repository.BookingReporitory;
+import com.example.hbs.repository.BookingRepository;
+import com.example.hbs.repository.HotelRepository;
+import com.example.hbs.repository.UserRepository;
+import com.example.hbs.service.mapper.BookingMapper;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
 @Service
 public class BookingService {
     @Autowired
-    BookingReporitory bookingReporitory;
+    BookingRepository bookingRepository;
 
-    public ResponseEntity<String> addBooking(Booking booking) {
-        bookingReporitory.save(booking);
-        return ResponseEntity.ok("Booking Added");
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    HotelRepository hotelRepository;
+
+    @Autowired
+    BookingMapper bookingMapper;
+
+    public BookingResponseDto addBooking(BookingRequestDto bookingRequestDto) {
+        if (userRepository.findById(bookingRequestDto.getUserId()).isPresent() &&
+                hotelRepository.findById(bookingRequestDto.getHotelId()).isPresent()) {
+            Booking booking = Booking.builder()
+                    .user(userRepository.findById(bookingRequestDto.getUserId()).get())
+                    .hotel(hotelRepository.findById(bookingRequestDto.getHotelId()).get())
+                    .checkIn(bookingRequestDto.getCheckIn())
+                    .checkOut(bookingRequestDto.getCheckOut())
+                    .bookingPrice(bookingRequestDto.getBookingPrice())
+                    .noOfRooms(bookingRequestDto.getNoOfRooms())
+                    .noOfPeople(bookingRequestDto.getNoOfPeople())
+                    .build();
```

Comment:
need some logic here to check if the hotel contains required no of rooms available, and decrement available rooms also